### ISSUE TITLE
[Bug]: 6.8.0 Link "to" prop partial Path value results in different output url

### ISSUE THREAD

**Issue by mmaloon** [remix-run/react-router#10007]
### What version of React Router are you using?

react-router-dom 6.8.0

### Steps to Reproduce

When browser is at a route other than the homepage: eg: `/about`, passing a Path object without the pathname key results in different final href values for 6.8.0 vs 6.7.0:

In the examples, click on the "About" link then inspect the "Hash Link" and "Search Link" to see the different `href` valeus

```jsx
// on page with route "/about"
<Link to={{ hash: 'hash' }}>Hash Link</Link>
<Link to={{ search: 'foo=bar' }}>Search Link</Link>
```
6.8.0 output: [example](https://stackblitz.com/edit/github-8n265p?file=src/App.tsx)
```
<a href="/#hash">Hash Link</a>
<a href="/?foo=bar">Search Link</a>
```

6.7.0 output: [example](https://stackblitz.com/edit/github-8n265p-lte6j2?file=src/App.tsx)
```
<a href="/about#hash">Hash Link</a>
<a href="/about?foo=bar">Search Link</a>
```

### Expected Behavior

Result in same final href value on 6.8.0 as in 6.7.0

### Actual Behavior

final href value is different on 6.8.0 than 6.7.0, resulting in a breaking change

**Comment by frankalbenesius** [comment]
This also affected my company's application. We have hotfixed a downgrade. Thank you for making this post @mmaloon .

**Comment by brophdawg11** [comment]
Thanks for the bug report!  Pushed a fix for this in the above PR 👍

**Comment by brophdawg11** [comment]
This is fixed in `6.8.1-pre.0` if you want to give that spin to confirm.  If all goes well we should have the stable release out tomorrow or early next week.

**Comment by brophdawg11** [comment]
This is now released in `6.8.1`

**Comment by mmaloon** [comment]
confirmed, issue resolved in `6.8.1`. Thanks!
